marshal.go

```diff
@@ -133,49 +134,74 @@ func marshalImpl(vo reflect.Value, w io.Writer, options *MarshalOptions) error {
 
 			for _, vn := range lst {
 				// Write element ID
-				if _, err := w.Write(e.b); err != nil {
-					return err
+				var headerSize uint64
+				if n, err := w.Write(e.b); err != nil {
+					return pos, err
+				} else {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)